nifi-extension-bundles/nifi-poi-bundle/nifi-poi-services/src/main/java/org/apache/nifi/excel/ExcelHeaderSchemaStrategy.java

```diff
@@ -85,16 +85,18 @@ public RecordSchema getSchema(Map<String, String> variables, InputStream content
         int index = 0;
 
         while (rowIterator.hasNext()) {
-           Row row = rowIterator.next();
-           if (index == 0) {
-               fieldNames = getFieldNames(firstRow, row);
-           } else if (index <= NUM_ROWS_TO_DETERMINE_TYPES) {
-               inferSchema(row, fieldNames, typeMap);
-           } else {
-               break;
-           }
-
-           index++;
+            Row row = rowIterator.next();
+            if (index == 0) {
+                fieldNames = getFieldNames(firstRow, row);
+            } else if (row.getRowNum() == zeroBasedFirstRow) { // skip first row of all sheets
```

Comment:
What do you think about clarifying the `USE_STARTING_ROW` description to note that the header is skipped in each worksheet, but the schema is inferred from only the first worksheet?

Reply:
That is a great suggestion! I will take care of that.